cloudinit/config/cc_yum_add_repo.py

```diff
@@ -211,7 +211,7 @@ def handle(name: str, cfg: Config, cloud: Cloud, args: list) -> None:
                 n_repo_config[k] = v
         repo_config = n_repo_config
         missing_required = 0
-        for req_field in ["baseurl"]:
+        for req_field in ["baseurl", "metalink"]:
```

Comment:
This will not do it unfortunately. "metalink" is optional as long as "baseurl" is present. "baseuril" is optional as long as "metalink" is present.